### ISSUE TITLE
Decide if we should add a "Reset password" link to the admin login page

### ISSUE THREAD

**Issue by angela-tran** [cal-itp/benefits#2386]
Apparently Django admin provides a feature for native Django accounts to reset their password.

https://docs.djangoproject.com/en/5.1/ref/contrib/admin/#adding-a-password-reset-feature

We have not tried it out yet and have no idea what it looks like.

This ticket is to decide if we want to enable it.

- [x] A developer should try out enabling it and see how it works
- [x] Share info as a comment on this ticket
- [x] Make sure @indexing agrees if we should add this or not

## Additional context
> I don't think we should worry about adding this now since we ideally want to use SSO long-term

_Originally posted by @angela-tran in https://github.com/cal-itp/benefits/pull/2371#discussion_r1767619933_

**Comment by angela-tran** [comment]
So in addition to copying the URL patterns from "[Adding a password reset feature](https://docs.djangoproject.com/en/5.1/ref/contrib/admin/#adding-a-password-reset-feature)" into `benefits/urls.py`, we would need an SMTP email server for this feature to work. We would specify the email server details in `benefits/settings.py` using the [settings for the SMTP email backend](https://docs.djangoproject.com/en/5.1/topics/email/#django.core.mail.backends.smtp.EmailBackend). Django has other implementations of email backends, but they are for development and testing purposes.

Here's what the screens in Django admin look like:

### Login screen
I had to hover over the link to see it.

![image](https://github.com/user-attachments/assets/cbae7b69-c5f3-4134-b5c3-8d502ae28634)

### Password reset form

![image](https://github.com/user-attachments/assets/ac8caddc-e83c-4333-9d4d-96982ef63b64)


### Password reset form submitted

![image](https://github.com/user-attachments/assets/28e51f03-a839-4f8f-a6b5-303496fb7e58)



The email contents by default come from this template: https://github.com/django/django/blob/39de2e97a06d0317973b280bc159ca6f89fc64e3/django/contrib/admin/templates/registration/password_reset_email.html

I couldn't get the development console-based email backend to work, so couldn't continue any further than this.

**Comment by angela-tran** [comment]
@indexing @thekaveman What are your thoughts about this feature based on the notes above?

**Comment by indexing** [comment]
What is the lift to bring up an SMTP email server?

**Comment by angela-tran** [comment]
> What is the lift to bring up an SMTP email server?

One option could be to use the @cal-itp-bot  Gmail account. Or set up a separate bot/service account for Benefits admin specifically.

Then, according to this [article](https://medium.com/repped-in-tech/setting-up-a-simple-email-connection-in-django-with-gmail-d577916be5bc), we would generate an app password for that account in Gmail, and then set these settings:
```
EMAIL_BACKEND = "django.core.mail.backends.smtp.EmailBackend"
EMAIL_HOST = "smtp.gmail.com"
EMAIL_PORT = 587
EMAIL_USE_TLS = True
EMAIL_HOST_USER = env("EMAIL_HOST_USER")
EMAIL_HOST_PASSWORD = env("EMAIL_HOST_PASSWORD")
```
where `EMAIL_HOST_USER` is the service email and `EMAIL_HOST_PASSWORD` is the generated app password.

**Comment by angela-tran** [comment]
I've been looking into the generated app password approach with @cal-itp-bot 's Gmail account. Looks like `App passwords` is [not an option](https://knowledge.workspace.google.com/kb/how-to-create-app-passwords-000009237)) if your account is through [work, school, or other organization](https://support.google.com/accounts/answer/6250166?hl=en).

I think this is why I can't generate an app password.

**Comment by angela-tran** [comment]
I also tried using the actual password for `EMAIL_HOST_PASSWORD`, but kept getting back a "Bad credentials" error when trying to get Django to send me a "reset password" email.

**Comment by angela-tran** [comment]
Ok, so I found another approach to try based on 
- https://stackoverflow.com/questions/72396092/how-can-i-send-email-using-python-with-google-workspace
- https://medium.com/@adithya.webdev/using-g-suite-account-to-send-emails-in-python-django-45f134ee1688

Basically, you create a service account in Google Cloud Console that has domain-wide delegation scoped just to `https://www.googleapis.com/auth/gmail.send` and download a JSON credential file to use as authentication. This allows the service account to impersonate sending email from an account in the domain.

The author of that Medium article created a package [`django_gsuite_email`](https://github.com/slicefox/django-gsuite-email) that lets you use this approach as your Django email backend.

`settings.py`
```python
INSTALLED_APPS = [
    ...
    'django_gsuite_email',
    ...
]
...

EMAIL_BACKEND = 'django_gsuite_email.GSuiteEmailBackend'
GSUITE_CREDENTIALS_FILE="/path/to/credentials/file.json"
```

I'm not able to set domain-wide delegation on the service account I created though because I'm not a workspace admin. I've disabled the service account I created.
